### DISCUSSION TITLE
Add friendly name to each SmartEvse device

### DISCUSSION THREAD

**Discussion by negronoir** [dingo35/SmartEVSE-3.5#28] [Ideas]
Hello. Thanks for the project

Could be possible to allow a 'label configuration screen' (may be on the display or even in the web server) to label the SmartEVSE with a friendly name?

The device name sent to e.g. HA by REST or MQTT is something like SmartEvse-12345. And on the web server that label doesn't even appear

If you have more than one SmartEVSE as my case, it's difficult to identify which SmartEVSE I'm configuring. I have to distinguish them by IP address

It could be a good idea to label SmartEVSE and when the web interface loads, it can display e.g. "SmartEVSE-Tesla" or "SmartEvse-Dingo" or whatever you wants. And also send that name via REST/MQTT

**Comment by dingo35** [comment]
Hi Negronoir,

The  number you saw might have been very confusing, that is why in the current development version this number has standardized now to the serial number of the device (which is on the sticker). We need a unique identifier because multiple SmartEVSE's can be connected in the same modbus network, and they also need to be uniquely identified in MQTT and HomeAssistant. The serial number is the one thing that is unique.

You will be happy to see that in the new Webserver screen, on the third row the host name is now presented, in my case:
"SmartEVSE-6498".

This is now also the hostname on your lan, so you can use:
ping smartevse-6498.lan

and/or 

ping smartevse-6498.local,    depending on your mDNS configuration.

We will not introduce a "user friendly" name, because then we lose control over the uniqueness of the hostname on the network, it would obfuscate code and make auto-discovery features (e.g. HomeAssistant) less automated.

Thanks for your input!



"negronoir" ***@***.***> schreef op 7 maart 2024 om 07:04:
